minidump-common/src/format.rs

```diff
@@ -1446,37 +1460,133 @@ multi_structs! {
     /// This struct matches the struct of the same name from minidumpapiset.h.
-    pub struct MINIDUMP_MISC_INFO4 {
+    pub struct MINIDUMP_MISC_INFO_4 {
         pub build_string: [u16; 260], // MAX_PATH
         pub dbg_bld_str: [u16; 40],
     }
+
+    // Includes fields from MINIDUMP_MISC_INFO..4
+    /// Miscellaneous process and system information
+    ///
+    /// This struct matches the struct of the same name from minidumpapiset.h.
+    pub struct MINIDUMP_MISC_INFO_5 {
+        pub xstate_data: XSTATE_CONFIG_FEATURE_MSC_INFO,
+        pub process_cookie: u32,
+    }
+    // TODO: read xstate_data and process the extra XSAVE sections at the
+    // end of each thread's cpu context.
 }
 
-//TODO: MINIDUMP_MISC_INFO_5
-/*
-typedef struct _MINIDUMP_MISC_INFO_5 {
-    ULONG32 SizeOfInfo;
-    ULONG32 Flags1;
-    ULONG32 ProcessId;
-    ULONG32 ProcessCreateTime;
-    ULONG32 ProcessUserTime;
-    ULONG32 ProcessKernelTime;
-    ULONG32 ProcessorMaxMhz;
-    ULONG32 ProcessorCurrentMhz;
-    ULONG32 ProcessorMhzLimit;
-    ULONG32 ProcessorMaxIdleState;
-    ULONG32 ProcessorCurrentIdleState;
-    ULONG32 ProcessIntegrityLevel;
-    ULONG32 ProcessExecuteFlags;
-    ULONG32 ProtectedProcess;
-    ULONG32 TimeZoneId;
-    TIME_ZONE_INFORMATION TimeZone;
-    WCHAR   BuildString[MAX_PATH];
-    WCHAR   DbgBldStr[40];
-    XSTATE_CONFIG_FEATURE_MSC_INFO XStateData;
-    ULONG32 ProcessCookie;
-} MINIDUMP_MISC_INFO_5, *PMINIDUMP_MISC_INFO_5;
-*/
+/// A descriptor of the XSAVE context which can be found at the end of
+/// each thread's cpu context.
+///
+/// The sections of this context are dumps of some of the CPUs registers
+/// (e.g. one section might contain the contents of the SSE registers).
+///
+/// Intel documents its XSAVE entries in Volume 1, Chapter 13 of the
+/// "Intel 64 and IA-32 Architectures Software Developer’s Manual".
+#[derive(Debug, Clone, Pread, SizeWith)]
+pub struct XSTATE_CONFIG_FEATURE_MSC_INFO {
+    /// The size of this struct.
+    pub size_of_info: u32,
+    /// The size of the XSAVE context.
+    pub context_size: u32,
+    /// The bit `enabled_features[i]` indicates that `features[i]` contains valid data.
+    pub enabled_features: u64,
+    /// The offset and size of each XSAVE entry inside the XSAVE context.
+    pub features: [XSTATE_FEATURE; 64],
+}
+
+impl Default for XSTATE_CONFIG_FEATURE_MSC_INFO {
+    fn default() -> Self {
+        Self {
+            size_of_info: std::mem::size_of::<XSTATE_CONFIG_FEATURE_MSC_INFO>() as u32,
+            context_size: 0,
+            enabled_features: 0,
+            features: [XSTATE_FEATURE::default(); 64],
+        }
+    }
+}
+
+impl XSTATE_CONFIG_FEATURE_MSC_INFO {
+    /// Gets an iterator of all the enabled features.
+    pub fn iter(&self) -> XstateFeatureIter {
+        XstateFeatureIter { info: self, idx: 0 }
+    }
+}
+
+/// An iterator of all the enabled features in an XSTATE_CONFIG_FEATURE_MSC_INFO.
+#[derive(Debug)]
+pub struct XstateFeatureIter<'a> {
```

Comment:
This structure is spelled `Xstate` vs the index enum that is spelled `XState`. Even though it's just a minor difference, it would be great consolidating this.